libopae/pluginmgr.c

```diff
@@ -133,6 +171,18 @@ STATIC int opae_plugin_mgr_configure_plugin(opae_api_adapter_table *adapter,
 	return cfg(adapter, config);
 }
 
+STATIC void opae_plugin_mgr_reset_cfg(void)
+{
+	int i = 0;
+	for ( ; i < plugin_count; ++i) {
+		if (plugin_list[i].cfg) {
```

Comment:
Rather than maintaining a separate list, does it make sense to put this data into the adapter?

Reply:
Can you elaborate? I maintain a separate list because I separate parsing/loading.

Reply:
I think I see.  It's a chicken and egg.  You need to read the config first before you know which adapter to create.
```c
typedef struct _plugin_cfg {
	char name[PLUGIN_NAME_MAX];
	char module[PLUGIN_NAME_MAX];
	bool enabled;
	char *cfg;
	size_t cfg_size;
	uint32_t supported_devices[PLUGIN_SUPPORTED_DEVICES_MAX];
} plugin_cfg;
```
I guess I was wondering whether it made sense to store "cfg" and "size" in the adapter?